crates/node/builder/src/launch/common.rs

```diff
@@ -75,9 +78,14 @@ impl LaunchContext {
 
         if !config.network.trusted_peers.is_empty() {
             info!(target: "reth::cli", "Adding trusted nodes");
-            config.network.trusted_peers.iter().for_each(|peer| {
-                toml_config.peers.trusted_nodes.insert(*peer);
-            });
+
+            // resolve trusted peers if they use a domain instead of dns
+            for peer in &config.network.trusted_peers {
+                let resolved = resolve_dns_node_record(peer.clone())
```

Comment:
Are we sure we want to resolve DNS at this point? Being able to do lazy DNS resolution might be preferred.

When deploying geth on k8s, we have found its behaviour to eagerly resolve DNS of hostnames in enode strings to be a blocker for trusted peer setups (e.g. trying to resolve hostname for a clusterip of a statefulset that is being deployed in parallel, causing all peers always crash).

Reply:
to make it work for this use case, what would be the best way to resolve?

Reply:
The hostname for a statefulset pod on k8s should stabilize within a few seconds of the pod being up and running. The issue with geth was that the pods would terminate almost instantly on start up due to the hostname resolution failure.

To give us the best shot of not having the same issue, we could consider some or all of these options:
1. Build in a configurable number of retries (and maybe backoff strategy) for resolving peer hostnames; 
3. Make the type for accessing peer records handle hostname resolution just in time (e.g. `peer.ip_addr()` would perform resolution if it had not been done before).

Option 1 would allow operators to ensure start up does not fail on the basis of hostnames that are in the process of being available on cluster.

Option 2 is not guaranteed to solve the issue, but might mitigate the number of retries required.

Let me know if you think we should address this in a separate issue and PR. And if you want me to help finish off this PR at all.